File: API/Data/Entities/Color.cs
using System;
using System.Collections.Generic;

namespace API.Data.Entities
{
    public partial class Color
    {
        public Color()
        {
            ProductColors = new HashSet<ProductColor>();
        }

        public int Id { get; set; }
        public string Name { get; set; } = null!;
        public string ColorCode { get; set; } = null!;
        public DateTime? CreateAt { get; set; }
        public DateTime? UpdateAt { get; set; }

        public virtual ICollection<ProductColor> ProductColors { get; set; }
    }
}

File: Client/src/app/shared/models/color.ts
import { SizeVariant } from "./size"

export interface Color {
    id: number
    name: string
    colorCode: string
}

export interface ColorVariant {
    color: string
    colorCode: string
    sizeVariants?: SizeVariant[]
}

export interface ColorDetail {
    id: number
    name: string
    colorCode: string
    createAt: string
    updateAt: string
    productCount: number
}